lib/active_interactor/deprecation.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module ActiveInteractor
+  # ActiveSupport::Deprecation instances used for deprecation logging.
+  #
+  # @see https://api.rubyonrails.org/classes/ActiveSupport/Deprecation.html ActiveSupport::Deprecation
+  #
+  # @author Aaron Allen <hello@aaronmallen.me>
+  # @since unreleased
+  module Deprecation
+    class << self
+      # Disable deprecation debuging
+      def disable_debugging!
+        each_deprecator { |deprecator| deprecator.debug = false }
+      end
+
+      # Disable deprecation logging
+      def disable_logging!
+        each_deprecator { |deprecator| deprecator.silenced = true }
+      end
+
+      # Enable deprecation debuging
+      def enable_debugging!
+        each_deprecator { |deprecator| deprecator.debug = true }
+      end
+
+      # Enable deprecation logging
+      def enable_logging!
+        each_deprecator { |deprecator| deprecator.silenced = false }
+      end
+
+      private
+
+      def deprecator(deprecation_horizion)
```

Comment:
Is this suppose to be `deprecation_horizon`?

Reply:
aye, it's a one-to-one mapping of the `ActiveSupport::Deprecation` initializer: https://api.rubyonrails.org/classes/ActiveSupport/Deprecation.html#method-c-new

Reply:
lol good catch, words are hard